libs/shared/flagd-core/src/lib/targeting/sem-ver.ts

```diff
@@ -0,0 +1,46 @@
+import {compare, parse} from 'semver'
+
+export const semVerRule = 'sem_ver';
+
+export function semVer(data: unknown): boolean {
+  if (!Array.isArray(data)) {
+    return false;
+  }
+
+  const args = Array.from(data)
+
+  if (args.length != 3) {
+    return false;
+  }
+
+  const semVer1 = parse(args[0]);
+  const semVer2 = parse(args[2]);
```

Comment:
Can this parsing fail?

Reply:
A `null` is returned if parsing failed, which we check in our logic